dotCMS/src/main/java/com/dotcms/security/apps/SecretsKeyStoreHelper.java

```diff
@@ -224,41 +200,36 @@ public boolean containsKey(final String variableKey){
         try {
            final KeyStore keyStore = getSecretsStore();
             if (keyStore.containsAlias(variableKey)) {
-                final PasswordProtection keyStorePP = new PasswordProtection(loadStorePassword());
+                final PasswordProtection keyStorePP = new PasswordProtection(passwordSupplier.get());
                 final SecretKeyFactory factory = SecretKeyFactory.getInstance(SECRETS_STORE_SECRET_KEY_FACTORY_TYPE);
                 final SecretKeyEntry secretKeyEntry = (SecretKeyEntry) keyStore.getEntry(variableKey, keyStorePP);
                 return ((PBEKeySpec) factory.getKeySpec(secretKeyEntry.getSecretKey(), PBEKeySpec.class)).getPassword();
             } else {
-                return CACHE_404;
+                return null;
```

Comment:
is it safe to return null here?

Reply:
yeah, pretty sure. it is handled in an upper layer. where if it is null I return the CACHE_404